sycl/doc/extensions/experimental/sycl_ext_oneapi_reduction_properties.asciidoc

```diff
@@ -145,7 +157,7 @@ request for specific reduction behavior(s), the implementation must throw an
 
```

Comment:
Do we need this paragraph:

> If a reduction kernel is submitted to a device that cannot satisfy the
request ...

?

That makes it sound like a valid implementation can just throw an exception for any of the reduction functions.  I don't think we want to allow that.

What problem are you trying to solve here?

Reply:
In general, I think we need to be more rigorous about optional features.  If we can't guarantee that a feature will work on all devices, we should have a query.  I think it's a bad API design if applications need to set up exception handlers in order to determine if a feature is supported.

It sounds like this paragraph is here mostly for future properties.  If that's the case, let's remove the paragraph for now, and we can assess whether future property need to be optional with a matching query.

Reply:
We should continue this discussion more offline, but briefly: I think there are some cases (like reductions) where `try`/`catch` might not be too bad as an API design, because the only real use-case is something like "if you can't give me this precise configuration, fall back to this manual unoptimized implementation".  That's different to some of the other use-cases where developers might actually want to run some orthogonal feature queries to make hierarchical decisions.

Regardless, I agree we should remove this for now, so I've done that in 3ea5c474c6df50c0ee62fb2ffa7023203d49ad65.